crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -2779,6 +2778,39 @@ impl AuthorityPerEpochStore {
         Ok(transactions_to_schedule)
     }
 
+    // Adds the consensus commit prologue transaction to the beginning of input `transactions` to update
+    // the system clock used in all transactions in the current consensus commit.
+    fn add_consensus_commit_prologue_transaction(
+        &self,
+        batch: &mut DBBatch,
+        transactions: &mut VecDeque<VerifiedExecutableTransaction>,
+        consensus_commit_info: &ConsensusCommitInfo,
+        roots: &mut BTreeSet<TransactionKey>,
+    ) -> SuiResult {
+        #[cfg(any(test, feature = "test-utils"))]
+        {
+            if consensus_commit_info.skip_consensus_commit_prologue_in_test() {
+                return Ok(());
+            }
+        }
+
+        let transaction = consensus_commit_info
+            .create_consensus_commit_prologue_transaction(self.epoch(), self.protocol_config());
+        match self.process_consensus_system_transaction(&transaction) {
+            ConsensusCertificateResult::SuiTransaction(processed_tx) => {
+                roots.insert(processed_tx.key());
+                self.record_consensus_message_processed(
+                    batch,
+                    SequencedConsensusTransactionKey::System(*processed_tx.digest()),
+                )?;
+                transactions.push_front(processed_tx);
+                Ok(())
+            }
+            ConsensusCertificateResult::IgnoredSystem => Ok(()),
```

Comment:
I am not quite sure what this means. So `transactions` won't include the prologue in the first transaction for this case?